source/application/utils/file/Path.cpp

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright(c) 2017 - Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#include "../File.h"
+
+std::string Utils::File::GetFileName(const std::string path)
```

Comment:
I've been thinking about this, what about a class called `Path` that inherits `std::string`?